src/core.py

```diff
@@ -6,44 +6,60 @@
 from pytesseract import Output
 import csv
 
-#Reading the image from path
-img = cv2.imread(((os.path.dirname(os.path.realpath(__file__))+"\\images\\image.jpg")))
-results = []
 
 # get grayscale image
 def get_grayscale(image):
     return cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
 
 # noise removal
+# CH: not sure why you're writing a function for what is just a single open CV function
+# All you do is hide which kernel size and blur method is used
 def remove_noise(image):
     return cv2.medianBlur(image,3)
  
+# If you really want a clean yet flexible function interface, I suggest you expose as many args as possible 
+# and use your currently hardcode values as defaults. You should also (later) add doc strings ....
+def remove_noise_CH(image, size=3):
+    '''removes noise by running a median blur filter with a given kernel size''' 
+    return cv2.medianBlur(image, size)
+
 #thresholding
 def thresholding(image):
     #return cv2.threshold(image, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)[1]
     return cv2.adaptiveThreshold(img,255,cv2.ADAPTIVE_THRESH_GAUSSIAN_C,\
             cv2.THRESH_BINARY,11,10)
 
+#thresholding
+def thresholding_CH(image, blocksize=11, constant=10):
+    #return cv2.threshold(image, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)[1]
+    return cv2.adaptiveThreshold(img, 255, cv2.ADAPTIVE_THRESH_GAUSSIAN_C,  # you don't need \ here b/c anything similar to
+                                 cv2.THRESH_BINARY,                         # a list, including args, are auto protected in Python
+                                 blockSize=blocksize,   # if you put one arg per line you can easily add a comment
+                                 C=constant)     # e.g.: subtracted from the mean or weighted sum of the neighbourhood pixels.
+
+# BTW I'm unclear where you got your blocksize and constant values from(?) Are these accepted good values or did you 
+# have to experiment with your specific type of input?
```

Comment:
I tried all the methods for threshold and these are what worked better with the images.